packages/core/src/components/alert/alert.tsx

```diff
@@ -122,97 +115,98 @@ export interface AlertProps extends OverlayLifecycleProps, Props {
 
     /**
      * Handler invoked when the Alert is confirmed or canceled; see `onConfirm` and `onCancel` for more details.
      * First argument is `true` if confirmed, `false` otherwise.
      * This is an alternative to defining separate `onConfirm` and `onCancel` handlers.
      */
-    onClose?(confirmed: boolean, evt?: React.SyntheticEvent<HTMLElement>): void;
+    onClose?: (confirmed: boolean, event?: React.SyntheticEvent<HTMLElement>) => void;
 }
 
 /**
  * Alert component.
  *
  * @see https://blueprintjs.com/docs/#core/components/alert
  */
-export class Alert extends AbstractPureComponent<AlertProps> {
-    public static defaultProps: AlertProps = {
-        canEscapeKeyCancel: false,
-        canOutsideClickCancel: false,
-        confirmButtonText: "OK",
-        isOpen: false,
-        loading: false,
-    };
-
-    public static displayName = `${DISPLAYNAME_PREFIX}.Alert`;
-
-    public render() {
-        const {
-            canEscapeKeyCancel,
-            canOutsideClickCancel,
-            children,
-            className,
-            icon,
-            intent,
-            loading,
-            cancelButtonText,
-            confirmButtonText,
-            onClose,
-            ...overlayProps
-        } = this.props;
-        return (
-            <Dialog
-                {...overlayProps}
-                role="alertdialog"
-                className={classNames(Classes.ALERT, className)}
-                canEscapeKeyClose={canEscapeKeyCancel}
-                canOutsideClickClose={canOutsideClickCancel}
-                onClose={this.handleCancel}
-            >
-                <div className={Classes.ALERT_BODY}>
-                    <Icon icon={icon} size={40} intent={intent} />
-                    <div className={Classes.ALERT_CONTENTS}>{children}</div>
-                </div>
-                <div className={Classes.ALERT_FOOTER}>
-                    <Button loading={loading} intent={intent} text={confirmButtonText} onClick={this.handleConfirm} />
-                    {cancelButtonText && (
-                        <Button text={cancelButtonText} disabled={loading} onClick={this.handleCancel} />
-                    )}
-                </div>
-            </Dialog>
-        );
-    }
-
-    protected validateProps(props: AlertProps) {
-        if (props.onClose == null && (props.cancelButtonText == null) !== (props.onCancel == null)) {
+export const Alert: React.FC<AlertProps> = props => {
+    const {
+        cancelButtonText,
+        canEscapeKeyCancel = false,
+        canOutsideClickCancel = false,
+        children,
+        className,
+        confirmButtonText = "OK",
+        icon,
+        intent,
+        isOpen = false,
+        loading = false,
+        onCancel,
+        onClose,
+        onConfirm,
+        ...overlayProps
+    } = props;
+
+    React.useEffect(() => {
+        if (onClose == null && (cancelButtonText == null) !== (onCancel == null)) {
```

Comment:
See my comment on spinner changes: https://github.com/palantir/blueprint/pull/7181/files#r1920669985

Reply:
Changed in 01c9f2ba72e200ee697027a04c3a57bee089bdf6